### PR TITLE
Updating readme with temperature in place of serialNumber

### PR DESCRIPTION
**Pull request by ericmitt**

### CODE REVIEW

File: iothub/device/samples/convention-based-samples/readme.md [hunk 1]
```diff
@@ -79,11 +79,11 @@ The following section provides a comparison between the older non-convention-awa
 #### Using non-convention-aware API (old):
 
 ```csharp
-// Send telemetry "serialNumber".
-string serialNumber = "SR-1234";
+// Send telemetry "temperature".
+int temperature = "70";
```
Comment (drwill-ms) [inline iothub/device/samples/convention-based-samples/readme.md:83]:
```suggestion
int temperature = 70;
```

File: iothub/device/samples/convention-based-samples/readme.md [hunk 2]
```diff
@@ -98,12 +98,12 @@ await _deviceClient.SendEventAsync(message, cancellationToken);
 #### Using convention-aware API (new):
 
 ```csharp
-// Send telemetry "serialNumber".
-string serialNumber = "SR-1234";
+// Send telemetry "temperature".
+int temperature = "70";
```
Comment (drwill-ms) [inline iothub/device/samples/convention-based-samples/readme.md:102]:
```suggestion
int temperature = 70;
```

File: iothub/device/samples/convention-based-samples/readme.md [hunk 3]
```diff
@@ -114,11 +114,11 @@ await _deviceClient.SendTelemetryAsync(telemetryMessage, cancellationToken);
 #### Using non-convention-aware API (old):
 
 ```csharp
-// Send telemetry "serialNumber" under component "thermostat1".
-string serialNumber = "SR-1234";
+// Send telemetry "temperature" under component "thermostat1".
+int temperature = "70";
```
Comment (drwill-ms) [inline iothub/device/samples/convention-based-samples/readme.md:118]:
```suggestion
int temperature = 70;
```

File: iothub/device/samples/convention-based-samples/readme.md [hunk 4]
```diff
@@ -134,12 +134,12 @@ await _deviceClient.SendEventAsync(message, cancellationToken);
 #### Using convention-aware API (new):
 
 ```csharp
-// Send telemetry "serialNumber" under component "thermostat1".
-string serialNumber = "SR-1234";
+// Send telemetry "temperature" under component "thermostat1".
+int temperature = "70";
```
Comment (drwill-ms) [inline iothub/device/samples/convention-based-samples/readme.md:138]:
```suggestion
int temperature = 70;
```

**Review by drwill-ms — CHANGES_REQUESTED** [review]
Type 'int' should not be initialized to string.

**Review by ericmitt — COMMENTED** [review]
better to have a double rather an int for the temperature

File: iothub/device/samples/convention-based-samples/readme.md [hunk 5]
```diff
@@ -79,11 +79,11 @@ The following section provides a comparison between the older non-convention-awa
 #### Using non-convention-aware API (old):
 
 ```csharp
-// Send telemetry "serialNumber".
-string serialNumber = "SR-1234";
+// Send telemetry "temperature".
+double temperature = 70;
```
Comment (drwill-ms) [inline iothub/device/samples/convention-based-samples/readme.md:83]: I see you switched this to a double. If you are going to do that (which I think is a good value for temperature), I think you should change the initialized value to something that doesn't look like an int.

IOW, it should be:

```suggestion
double temperature = 70.0D;
```

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**